clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -659,25 +623,25 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                                         keyByteArray == null ? ConsumerRecord.NULL_SIZE : keyByteArray.length,
                                         valueByteArray == null ? ConsumerRecord.NULL_SIZE : valueByteArray.length,
                                         key, value);
-        } catch (KafkaException e) {
+        } catch (InvalidRecordException e) {
             throw e;
```

Comment:
Yes, indeed. Jason suggested (offline) the same thing. He said he was going to push a commit with that change.